### PR TITLE
Assignment 4 - b19M057

### PR DESCRIPTION
**Pull request by pmad19**

### CODE REVIEW

**Comment by pmad19** [comment]
@dgarijo This is the new pull request for Assignment 4. I have separated the two commits in two pull request as you commented me in https://github.com/FacultadInformatica-LinkedData/Curso2023-2024-ODKG/pull/204. I have also close the old pull request https://github.com/FacultadInformatica-LinkedData/Curso2023-2024-ODKG/pull/204 and https://github.com/FacultadInformatica-LinkedData/Curso2023-2024-ODKG/pull/285

File: Assignment4/pmad19-b19M057/task07.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+ns = Namespace("http://somewhere#")
+
+# With RDFLib
+subclasses_rdflib = [s for s,p,o in g.triples((None, RDFS.subClassOf, ns.LivingThing))]
+
+# With SPARQL
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE{
+    ?Subject rdfs:subClassOf ns:LivingThing
```
Comment (dgarijo) [inline Assignment4/pmad19-b19M057/task07.py:37]: this does not retrieve all subclasses of subclasses of LivingThing

File: Assignment4/pmad19-b19M057/task07.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+"""
+
+# TO DO
+# With RDFLib
+individuals = [s for s,p,o in g.triples((None, RDF.type, ns.Person))]
+subclasses = [s for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person))]
+for i in subclasses:
+  subclass_individuals = [s for s,p,o in g.triples((None, RDF.type, i))]
+  for individual in subclass_individuals:
+    individuals.append(individual)
```
Comment (dgarijo) [inline Assignment4/pmad19-b19M057/task07.py:63]: same as above

File: Assignment4/pmad19-b19M057/task07.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+# Visualize the results
+for i in individuals:
+  print(i)
+
+print("--------")
+
+for r in g.query(q2):
+  print(r.Subject)
+
+"""**TASK 7.3: List all individuals of "Person" and all their properties including their class with RDFLib and SPARQL**
```
Comment (dgarijo) [inline Assignment4/pmad19-b19M057/task07.py:84]: You are not doing the right exercises. This is taken from last year.

Comment (dgarijo) [reply Assignment4/pmad19-b19M057/task07.py:84]: 7.3, 7.4 and 7.5 missing.
Also, exercises are missing in rdflib

**Review by dgarijo — CHANGES_REQUESTED** [review]
Task 7 is incomplete

**Comment by SerPablo** [comment]
Errors not solved. Pull request closed. No more PR are allowed

**Pull request closed without merging**